examples/rust/bzsstack/src/main.rs

```diff
@@ -0,0 +1,211 @@
+use std::u64;
+
+use std::boxed::Box;
+use std::io::Error;
+use std::mem;
+use std::result::Result;
+use std::time::Duration;
+
+extern crate nix;
+use nix::unistd::close;
+
+extern crate libbpf_rs;
+
+extern crate clap;
+use clap::Parser;
+
+#[path = "bpf/.output/profile.skel.rs"]
+mod profile;
+use profile::*;
+
+extern crate blazesym;
+use blazesym::*;
+
+extern crate libc;
+
+mod syscall;
+
+const MAX_STACK_DEPTH: usize = 127;
+
+// A Rust version of stacktrace_event in profile.h
+#[repr(C)]
+struct stacktrace_event {
+    pid: u32,
+    cpu_id: u32,
+    kernel_stack_size: u32,
+    user_stack_size: u32,
+    kernel_stack: [u64; MAX_STACK_DEPTH],
+    user_stack: [u64; MAX_STACK_DEPTH],
+}
+
+fn init_perf_monitor(freq: u64) -> Vec<libc::c_int> {
+    let nprocs = libbpf_rs::num_possible_cpus().unwrap();
+    let pid = -1;
+    let buf: Vec<u8> = vec![0; mem::size_of::<syscall::perf_event_attr>()];
+    let mut attr = unsafe {
+        Box::<syscall::perf_event_attr>::from_raw(
+            buf.leak().as_mut_ptr() as *mut syscall::perf_event_attr
+        )
+    };
+    attr._type = syscall::PERF_TYPE_HARDWARE;
+    attr.size = mem::size_of::<syscall::perf_event_attr>() as u32;
+    attr.config = syscall::PERF_COUNT_HW_CPU_CYCLES;
+    attr.sample.sample_freq = freq;
+    attr.flags = 1 << 10; // freq = 1
+    (0..nprocs)
+        .map(|cpu| {
+            let fd = syscall::perf_event_open(attr.as_ref(), pid, cpu as i32, -1, 0);
+            fd as libc::c_int
```

Comment:
Is there a reason this is returned as libc::c_int instead of i32? It seems like returning this as a vector of i32s will make things easier (eg in attach_perf_event, perfds can be an &[i32] and not have to do any casting)?

Reply:
You are right!  The reason for casting to c_int is not valid anymore.